p2p_test.go

```diff
@@ -361,7 +361,7 @@ func TestBlockHeader_UnmarshalJSON(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			baseBlockFormat := `{"action_mroot":"","transaction_mroot":"","confirmed":0,"previous":"","producer":"","schedule_version":0,"timestamp":"0001-01-01T00:00:00","header_extensions":null%s}`
```

Comment:
I would like to ensure that the older format is still supported. Either in this test somehow or in another test dedicated to it (that maybe is already existing, just want to be sure there is no lost here).

Reply:
Then would it be possible that the response of the early block and recent block are different?

The responses of the mock server were captured from a couple of days ago.

Reply:
Would it be possible to define a version setter in the package root as below:

```go
type Version int

const (
    V1 EosioVersion = 1 + iota
    V2
)

var EosioVersion Version = V2

func SetVersion(ver Version) {
    EosioVersion = ver
}
```

and implement as below?:

```go
const blockTimestampFormat = "2006-01-02T15:04:05.999"

func (t BlockTimestamp) MarshalJSON() ([]byte, error) {
	strTime := t.Format(blockTimestampFormat)
	if len(strTime) == len("2006-01-02T15:04:05.5") {
		strTime += "00"
	} else if len(strTime) == len("2006-01-02T15:04:05") && EosioVersion == V2 {
		// modified since EOSIO/Leap v2.0
		// force formatting into YYYY-MM-DDTHH:MM:SS.sss
		strTime += ".000"
	}

	return []byte(`"` + strTime + `"`), nil
}
```

Reply:
@psy2848048?

Reply:
Sorry for confuse my previous comment which is deleted

Even I think that asserting "0001-01-01T00:00:00" and "0001-01-01T00:00:00.000" are not important, nor does "0" and "0.000000" but current `assert.JSONEq` cannot make it with tolerance. It seems that strict unmarshaling is not really productive if it is not related signing.

So I decide to delete a modified `BlockTimestamp` and `Int` & `Decimal`, and implement a kind of loosen JSONEq implementation - detect a real type from the string attribute, and convert into time and compare by Go time object if it types `Time`, comparing with rounded number if it types `Float`, and more

Reply:
I would just like to ensure we are accepting both form for unmarshalling. I know in the past that with and without `.000` bit us.

For signing purposes, data is packed using a deterministic binary encoding, so JSON representation is not important for signing in EOSIO world.

About the loose `JSONEq`, I think it's not something that should be pursue. Detecting real type from string is always hard. What other ways could we have to make the tests pass without implementing a specialized `JSONEq`? Maybe just changing a bit the actual input?

Reply:
Appreciated and manipulated the mok expected result if the diff of the actual and expected result are almost same and meaningless

https://github.com/eoscanada/eos-go/pull/191/commits/ca5b8435b0ca0323c521abc3d07eb6985015a0f6